src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -171,44 +171,54 @@ public async Task<ContentItem> GetAsync(string contentItemId, VersionOptions opt
             }
 
             ContentItem contentItem = null;
-
-            if (options.IsLatest)
-            {
-                contentItem = await _session
-                    .Query<ContentItem, ContentItemIndex>()
-                    .Where(x => x.ContentItemId == contentItemId && x.Latest == true)
-                    .FirstOrDefaultAsync();
-            }
-            else if (options.IsDraft && !options.IsDraftRequired)
+            if (options.IsAllVersions)
             {
                 contentItem = await _session
                     .Query<ContentItem, ContentItemIndex>()
                     .Where(x =>
-                        x.ContentItemId == contentItemId &&
-                        x.Published == false &&
-                        x.Latest == true)
+                        x.ContentItemId == contentItemId)
```

Comment:
Hmm, I think better to not take into account soft deleted items having both Published = Latest = false.

    x.ContentItemId == contentItemId && (x.Latest || x.Published)

Reply:
Not really, if you select AllVersion but it is still restricted. It's confusing

Reply:
> Hmm, I think better to not take into account soft deleted items having both Published = Latest = false.

@jtkech Yes, you are right, if you want to get different versions, you need to specify the version number.

Reply:
We need to have a DeletedVersions option then too I think.

Reply:
But the current content state is a little bit too complicated. For example, as a new person, you need to look up code or documentation to fully understand what the state means